moz_security/sandboxes/heka/analysis/moz_security_sshd_login_monitor.lua

```diff
@@ -36,24 +59,48 @@ local msg = {
 }
 
 function process_message()
-    local user    = read_message("Fields[user]")
-    local ip      = read_message("Fields[ssh_remote_ipaddr]")
+    local ts      = read_message("Timestamp")
+    local hn      = read_message("Hostname") or "unknown"
+    local user    = read_message("Fields[user]") or "unknown"
+    local ip      = read_message("Fields[ssh_remote_ipaddr]") or "unknown"
     local city    = read_message("Fields[ssh_remote_ipaddr_city]")
     local country = read_message("Fields[ssh_remote_ipaddr_country]")
 
-    msg.Fields[2].value    = string.format("%s logged into bastion from %s", user, ip)
-    -- If we also have city and country information, append that to the message
+    if not ts then
+        return -1, "event had no timestamp"
+    end
+    local tss = math.floor(ts / 1e9)
+
+    -- first, adjust the message for the default case
+    msg.Fields[2].value = string.format("%s bastion login %s from %s", user, hn, ip)
+    msg.Payload = string.format("Generated by %s, event timestamp %s\n",
+        cephost, os.date("%Y-%m-%d %H:%M:%S", tss))
+
+    -- if we also have city and country information, append that to the message subject
     if city and country then
         msg.Fields[2].value = msg.Fields[2].value .. string.format(" (%s, %s)", city, country)
     end
-    if not only_default then
-        msg.Fields[3].value[2] = string.format("<manatee-%s@moz-svc-ops.pagerduty.com>", user)
+
+    -- if user_email is set, add a user specific notification address
+    if user_email then
+        msg.Fields[3].value[2] = string.format(string.format("<%s>", user_email), user)
+    end
+
+    -- finally, if the message is delayed modify the recipient address and add some additional
+    -- information to the payload
+    local delaysec = os.time() - tss
+    if delaysec > acceptable_message_delay then
```

Comment:
We probably want throttling on the delay alerts

Reply:
They may contain useful information though, by throttle are you thinking drop until we exceed a certain time threshold than create again?

Reply:
This is a pretty low volume stream so it is not a must. I was just thinking if it did alert ops will investigate pretty quickly but there is no reason to ping them with more alerts during the investigation window.